api/app/Builders/UserBuilder.php

```diff
@@ -827,4 +829,45 @@ public function whereGeneralSearch(?string $searchTerm): self
 
         return $this;
     }
+
+    public function whereGeneralSearchBeta(?string $searchTerm): self
+    {
+        $queryTextPrefixMatch = PostgresTextSearch::searchStringToQueryText($searchTerm, PostgresTextSearchMatchingType::PREFIX);
+        $queryTextExactMatch = PostgresTextSearch::searchStringToQueryText($searchTerm, PostgresTextSearchMatchingType::EXACT);
+
+        if ($queryTextPrefixMatch && $queryTextExactMatch) {
+            $this
+                ->join('user_search_indices', 'users.id', '=', 'user_search_indices.id')
+                // attach the queries to every row to use for filtering
+                ->crossJoinSub(function ($query) use ($queryTextPrefixMatch, $queryTextExactMatch) {
+                    $query->selectRaw(
+                        'to_tsquery(coalesce(?, get_current_ts_config()), ?) AS prefix_match_query',
+                        ['english', $queryTextPrefixMatch]
+                    );
+                    $query->selectRaw(
+                        'to_tsquery(coalesce(?, get_current_ts_config()), ?) AS exact_match_query',
+                        ['english', $queryTextExactMatch]
+                    );
+                }, 'calculations')
+                // filter rows against the queries (OR)
+                ->where(function ($query) {
+                    $query->whereColumn('user_search_indices.searchable', '@@', 'calculations.prefix_match_query');
+                    $query->orWhereColumn('user_search_indices.searchable', '@@', 'calculations.exact_match_query');
+                })
+                // add the calculated rank column to allow for ordering by text search rank
+                // weighting the exact matches a little heavier
+                ->addSelect(DB::raw(<<<'SQL'
+                        (
+                            (ts_rank(user_search_indices.searchable, calculations.prefix_match_query) * 0.9) +
+                            (ts_rank(user_search_indices.searchable, calculations.exact_match_query) * 1.1)
+                        ) / 2 AS search_rank
+                        SQL))
+                // Now that we have added a column, query builder no longer will add a * to the select.  Add all possible columns manually.
+                ->addSelect(['users.*'])
```

Comment:
do we need users.* in here? we have only handful of columns in these tables.. can't we just select required columns?